Undefined names may raise NameError at runtime

flake8 testing of https://github.com/quantumlib/OpenFermion on Python 3.6.3

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./src/openfermion/ops/_interaction_rdm.py:53:28: F821 undefined name 'unpack_spatial_rdm'
one_rdm, two_rdm = unpack_spatial_rdm(one_rdm_a, one_rdm_b,
^

./src/openfermion/ops/_interaction_rdm.py:56:20: F821 undefined name 'constant'
return cls(constant, one_rdm, two_rdm)
^
```
https://github.com/quantumlib/OpenFermion/blob/master/src/openfermion/ops/_interaction_rdm.py#L50-L56

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -156,11 +156,6 @@ html_theme = 'sphinx_rtd_theme'
 #
 # html_favicon = None
 
-# Add any paths that contain custom static files (such as style sheets) here,
-# relative to this directory. They are copied after the builtin static files,
-# so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
-
 # Add any extra paths that contain custom files (such as robots.txt or
 # .htaccess) here, relative to this directory. These files are copied
 # directly to the root of the documentation.

--- a/docs/openfermion.rst
+++ b/docs/openfermion.rst
@@ -4,14 +4,14 @@ Code Documentation
 ==================
 
 openfermion.hamiltonians
---------------
+------------------------
 .. automodule:: openfermion.hamiltonians
     :members:
     :special-members: __init__
     :imported-members:
 
 openfermion.measurements
---------------
+------------------------
 
 .. automodule:: openfermion.measurements
     :members:
@@ -19,7 +19,7 @@ openfermion.measurements
     :imported-members:
 
 openfermion.ops
-------------
+---------------
 
 .. automodule:: openfermion.ops
     :members:
@@ -27,7 +27,7 @@ openfermion.ops
     :imported-members:
 
 openfermion.transforms
--------------------
+----------------------
 
 .. automodule:: openfermion.transforms
     :members:
@@ -35,7 +35,7 @@ openfermion.transforms
     :imported-members:
 
 openfermion.utils
---------------
+-----------------
 
 .. automodule:: openfermion.utils
     :members:

--- a/src/openfermion/ops/_interaction_rdm.py
+++ b/src/openfermion/ops/_interaction_rdm.py
@@ -50,14 +50,6 @@ class InteractionRDM(PolynomialTensor):
         self.one_body_tensor = self.n_body_tensors[1, 0]
         self.two_body_tensor = self.n_body_tensors[1, 1, 0, 0]
 
-    @classmethod
-    def from_spatial_rdm(cls, one_rdm_a, one_rdm_b,
-                         two_rdm_aa, two_rdm_ab, two_rdm_bb):
-        one_rdm, two_rdm = unpack_spatial_rdm(one_rdm_a, one_rdm_b,
-                                              two_rdm_aa, two_rdm_ab,
-                                              two_rdm_bb)
-        return cls(constant, one_rdm, two_rdm)
-
     def expectation(self, operator):
         """Return expectation value of an InteractionRDM with an operator.
 

--- a/src/openfermion/ops/_polynomial_tensor.py
+++ b/src/openfermion/ops/_polynomial_tensor.py
@@ -58,7 +58,7 @@ def two_body_basis_change(two_body_tensor, rotation_matrix):
 
     Procedure we use is an N^5 transformation which can be expressed as
     (pq|rs) = \sum_a R^p_a
-      (\sum_b R^q_b (\sum_c R^r_c (\sum_d R^s_d (ab|cd)))).
+    (\sum_b R^q_b (\sum_c R^r_c (\sum_d R^s_d (ab|cd)))).
 
     Args:
         two_body_tensor: a square rank 4 interaction tensor.

--- a/src/openfermion/utils/_operator_utils.py
+++ b/src/openfermion/utils/_operator_utils.py
@@ -39,28 +39,28 @@ def count_qubits(operator):
             PolynomialTensor, or InteractionRDM.
 
     Returns:
-        n_qubits (int): The minimum number of qubits on which operator acts.
+        num_qubits (int): The minimum number of qubits on which operator acts.
 
     Raises:
        TypeError: Operator of invalid type.
     """
     # Handle FermionOperator.
     if isinstance(operator, FermionOperator):
-        n_qubits = 0
+        num_qubits = 0
         for term in operator.terms:
             for ladder_operator in term:
-                if ladder_operator[0] + 1 > n_qubits:
-                    n_qubits = ladder_operator[0] + 1
-        return n_qubits
+                if ladder_operator[0] + 1 > num_qubits:
+                    num_qubits = ladder_operator[0] + 1
+        return num_qubits
 
     # Handle QubitOperator.
     elif isinstance(operator, QubitOperator):
-        n_qubits = 0
+        num_qubits = 0
         for term in operator.terms:
             if term:
-                if term[-1][0] + 1 > n_qubits:
-                    n_qubits = term[-1][0] + 1
-        return n_qubits
+                if term[-1][0] + 1 > num_qubits:
+                    num_qubits = term[-1][0] + 1
+        return num_qubits
 
     # Handle InteractionOperator, InteractionRDM, PolynomialTensor.
     elif isinstance(operator, (InteractionOperator,

--- a/src/openfermion/utils/_slater_determinants.py
+++ b/src/openfermion/utils/_slater_determinants.py
@@ -267,15 +267,17 @@ def givens_decomposition(unitary_rows):
     Args:
         unitary_rows: A numpy array or matrix with orthonormal rows,
             representing the matrix Q.
+
     Returns:
         left_unitary: An m x m numpy array representing the matrix V.
+
         givens_rotations: A list of tuples of objects describing Givens
-            rotations. The list looks something like
-            [(G_1, ), (G_2, G_3), ... ]. The Givens rotations within a tuple
-            can be implemented in parallel. The description of a Givens
-            rotation is itself a tuple of the form (i, j, theta, phi), which
-            represents a Givens rotation of rows i and j by angles theta
-            and phi.
+            rotations. The list looks like [(G_1, ), (G_2, G_3), ... ].
+            The Givens rotations within a tuple can be implemented in parallel.
+            The description of a Givens rotation is itself a tuple of the
+            form (i, j, theta, phi), which represents a Givens rotation of
+            rows i and j by angles theta and phi.
+
         diagonal: A list of the nonzero entries of D.
     """
     current_matrix = numpy.copy(unitary_rows)

--- a/src/openfermion/utils/_sparse_tools.py
+++ b/src/openfermion/utils/_sparse_tools.py
@@ -273,10 +273,11 @@ def jw_get_ground_states_by_particle_number(sparse_operator, particle_number,
         ground_energy(float): The lowest eigenvalue of sparse_operator within
             the eigenspace of the number operator corresponding to
             particle_number.
+
         ground_states(list[ndarray]): A list of the corresponding eigenstates.
 
-    Warning: The running time of this method is exponential in the number
-        of qubits.
+    Warning:
+        The running time of this method is exponential in the number of qubits.
     """
     # Check if operator is Hermitian
     if not is_hermitian(sparse_operator):

--- a/src/openfermion/utils/_unitary_cc.py
+++ b/src/openfermion/utils/_unitary_cc.py
@@ -78,14 +78,14 @@ def uccsd_operator(single_amplitudes, double_amplitudes, anti_hermitian=True):
 def uccsd_convert_amplitude_format(single_amplitudes, double_amplitudes):
     """Re-format single_amplitudes and double_amplitudes from ndarrays to lists.
 
-        Args:
+    Args:
         single_amplitudes(ndarray): [NxN] array storing single excitation
             amplitudes corresponding to t[i,j] * (a_i^\dagger a_j - H.C.)
-        double_amplitudes(ndarray): [NxNxNxN] array storing double excitation
-            amplitudes corresponding to
+        double_amplitudes(ndarray): [NxNxNxN] array storing double
+            excitation amplitudes corresponding to
             t[i,j,k,l] * (a_i^\dagger a_j a_k^\dagger a_l - H.C.)
 
-        Returns:
+    Returns:
         single_amplitudes_list(list): list of lists with each sublist storing
             a list of indices followed by single excitation amplitudes
             i.e. [[[i,j],t_ij], ...]

